src/components/marker/timeline-preview.tsx

```diff
@@ -0,0 +1,230 @@
+import {Component, Fragment, h, VNode} from 'preact';
+import * as styles from './timeline-preview.scss';
+import {ItemTypes, ThumbnailInfo} from "../../types/timelineTypes";
+import {Chapter, CuePointMarker} from "../../../flow-typed/types/cue-point-option";
+import {Icon, IconSize} from '@playkit-js/common/dist/icon';
+
+const {withText, Text} = KalturaPlayer.ui.preacti18n;
+
+const {
+  redux: {connect},
+  reducers: {seekbar},
+  components: {PLAYER_SIZE}
+} = KalturaPlayer.ui;
+
+const translates = {
+  questionTranslate: <Text id="timeline.question_title">Question</Text>,
+  reflectionPointTranslate: <Text id="timeline.reflection_point_title">Reflection Point</Text>,
+  hotspotTranslate: <Text id="timeline.hotspot_title">Hotspot</Text>,
+  aoaTranslate: <Text id="timeline.audience_asked_title">Audience Asked</Text>
+};
+
+interface TimelinePreviewProps {
+  onArrowClick: () => void,
+  cuePointsData: Array<CuePointMarker>,
+  thumbnailInfo: any,
+  isNavigationPluginOpen: () => boolean,
+  shouldRenderArrowButton: () => boolean,
+  questionTranslate?: string,
+  reflectionPointTranslate?: string,
+  hotspotTranslate?: string,
+  aoaTranslate?: string,
+  virtualTime?: number,
+  hoverActive?: boolean,
+  chapters?: Chapter[],
+  updateHoveredSegment?: (id: string, isHovered: boolean) => {},
+  playerSize?: string
+}
+
+const getFramePreviewImgContainerStyle = (thumbnailInfo: ThumbnailInfo) => {
+  return thumbnailInfo ? {
+    height: `${thumbnailInfo.height}px`,
+    width: `${thumbnailInfo.width}px`
+  } : {};
+};
+const getFramePreviewImgStyle = (thumbnailInfo: ThumbnailInfo) => {
+  if (thumbnailInfo) {
+    let framePreviewImgStyle = `height: 100%; width: 100%; background: url(${thumbnailInfo.url});`;
```

Comment:
styles tag works well with object, no need to convert in string

Reply:
i didn't address this one for now, as this part was a copy-paste from ivq old code and currently i don't feel like it's critical to change it